server/block/wall.go

```diff
@@ -0,0 +1,283 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/internal/sliceutil"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Wall is a block similar to fences that prevents players from jumping over and is thinner than the usual block. It is
+// available for many blocks and all types connect together as if they were the same type.
+type Wall struct {
+	transparent
+	// Block is the block to use for the type of wall.
+	Block world.Block
+	// NorthConnection is the type of connection in the north direction of the post.
+	NorthConnection WallConnectionType
+	// EastConnection is the type of connection in the east direction of the post.
+	EastConnection WallConnectionType
+	// SouthConnection is the type of connection in the south direction of the post.
+	SouthConnection WallConnectionType
+	// WestConnection is the type of connection in the west direction of the post.
+	WestConnection WallConnectionType
+	// Post is if the wall is extended to the full height of a block or not.
+	Post bool
+}
+
+// EncodeItem ...
+func (w Wall) EncodeItem() (string, int16) {
+	name, meta := encodeWallBlock(w.Block)
+	if meta == 0 {
+		return "minecraft:" + name + "_wall", 0
+	}
+	return "minecraft:cobblestone_wall", meta
+}
+
+// EncodeBlock ...
+func (w Wall) EncodeBlock() (string, map[string]any) {
+	properties := map[string]any{
+		"wall_connection_type_north": w.NorthConnection.String(),
+		"wall_connection_type_east":  w.EastConnection.String(),
+		"wall_connection_type_south": w.SouthConnection.String(),
+		"wall_connection_type_west":  w.WestConnection.String(),
+		"wall_post_bit":              boolByte(w.Post),
+	}
+	name, meta := encodeWallBlock(w.Block)
+	if meta > 0 || name == "cobblestone" {
+		properties["wall_block_type"] = name
+		name = "cobblestone"
+	}
+	return "minecraft:" + name + "_wall", properties
+}
+
+// Model ...
+func (w Wall) Model() world.BlockModel {
+	return model.Wall{
+		NorthConnection: w.NorthConnection.Height(),
+		EastConnection:  w.EastConnection.Height(),
+		SouthConnection: w.SouthConnection.Height(),
+		WestConnection:  w.WestConnection.Height(),
+		Post:            w.Post,
+	}
+}
+
+// BreakInfo ...
+func (w Wall) BreakInfo() BreakInfo {
+	hardness := 2.0
+	name, _ := encodeWallBlock(w.Block)
+	if name == "cobbled_deepslate" || name == "deepslate_brick" || name == "deepslate_tile" || name == "polished_deepslate" {
+		hardness = 3.5
+	}
```

Comment:
This should probably be in WallType

Reply:
The hardness for several wall types are also missing

Reply:
There is no WallType